gallery/src/styles/NxFormLayout/NxFormInlineLayoutExample.tsx

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { FormEvent } from 'react';
+
+import { NxButton } from '@sonatype/react-shared-components';
+import { NxStatefulTextInput } from '@sonatype/react-shared-components';
+
+export default function NxFormLayoutExample() {
+  function validator(val: string) {
+    return val.length ? null : 'Must be non-empty';
+  }
+
+  function onSubmit(evt: FormEvent) {
+    evt.preventDefault();
+    alert('Submitted!');
+  }
+
+  return (
+    <form className="nx-form" onSubmit={onSubmit}>
+      <div className="nx-form-group nx-form-group--inline">
+        <label className="nx-label">
+          <span className="nx-label__text">Label</span>
+          <NxStatefulTextInput validator={validator}/>
+          <NxButton variant="primary" type="submit">Submit</NxButton>
```

Comment:
You have the button inside the same `<label>` as the input?  Is that allowed/does it make sense?  Reading the standard, I don't think it is.

Reply:
> You have the button inside the same `<label>` as the input? Is that allowed/does it make sense? Reading the standard, I don't think it is.

Yeah, that wasn't allowed at all, I replaced it with a `flex box` based solution here and added an example that shows it working with a sub-title: https://github.com/sonatype/sonatype-react-shared-components/pull/164/commits/43f39c241667c908c92b52989983f40248909845